hack/generator/pkg/astmodel/type_definition.go

```diff
@@ -15,10 +15,18 @@ type TypeDefinition struct {
 	name        TypeName
 	description []string
 	theType     Type
+
+	// flags are used to identify resources during processing;
+	// they have no direct representation in the generated code.
+	flags map[Flag]struct{}
```

Comment:
You have flags on both `Resource` and `TypeDefinition` - but `TypeDefinition` can be for a `Resource`.

Are `flags` for `Type` implementations, or for `TypeDefinitions`, or for both?

What does it mean to have a `TypeDefinition` whose `Type` _also_ has a flag?

Would it make more sense to put some of the `Flag` methods on the `Type` interface? Or do we only allow them on resources?

Reply:
They're now on `TypeDefinition`, `ObjectType`, and `Resource`, with a level of code duplication that I dislike intensely.

Can't put the methods on the `Type` interface because the return type needs to be the implementation type, not `Type` - otherwise the fluent interface breaks down.

Aside: C# is finally going to address this by allowing covariant return types - as long as the implementation type is more specific, it will match the interface.